src/gather_classes.rs

```diff
@@ -183,42 +196,48 @@ fn process_compare_paths(
             100.00 * ((value as f64) / (total as f64)),
         )
     };
-    let mut failures = 0;
-    let mut processed = 0;
 
     let mut overall_report = overall_report;
-    // TODO (#72) Potential speedup: Process each file with `Process(path) -> ClassHashesReport` in parallel, then merge ClassHashesReport together.
-    for (block_num, path) in file_paths.into_iter() {
-        info!(
-            "{}\t{}\t{}",
-            get_stat("Processed", processed),
-            get_stat("Successes", processed - failures),
-            get_stat("Failures", failures)
-        );
-        info!(
-            "Processing path: `{}`",
-            path.file_name().unwrap().to_str().unwrap()
-        );
-        processed += 1;
+    let semaphore = Arc::new(Semaphore::new(max_block_processing_count));
+
+    let mut tasks = Vec::new();
+    for (block_num, path) in file_paths {
+        let sempahore_clone = Arc::clone(&semaphore);
+        let task = task::spawn(async move {
+            let _permit = sempahore_clone.acquire().await.unwrap();
+            process_single_file(block_num, path).await
+        });
+        tasks.push(task);
+    }
 
-        let json_result: Result<Value, _> = try_deserialize(path);
-        if let Err(err) = json_result {
-            warn!("Failed to process file with err: {err}");
-            failures += 1;
-            continue;
-        }
-        let json_obj = json_result.unwrap();
+    let results = future::join_all(tasks).await;
 
-        let update_result = update_report(&mut overall_report, &json_obj, block_num);
-        match update_result {
+    let mut processed = 0;
+    let mut failures = 0;
+
+    for result in results {
+        processed += 1;
+        match result {
             Err(err) => {
-                info!("{err}");
+                warn!("{err}");
+                failures += 1;
+            }
+            Ok(Err(err)) => {
+                warn!("{err}");
                 failures += 1;
             }
-            Ok(updates) => {
-                info!("Updates to overall report from block {block_num}: {updates}");
-                let _ = try_serialize(path_for_overall_report(network), &overall_report)
-                    .inspect_err(|err| warn!("{err}"));
+            Ok(Ok((block_num, report))) => {
```

Comment:
This looks odd, a result inside of a result doesn't make too much sense. Please update to remove this redundancy 
`Ok(Ok(T))` should just be `Ok(T)` 
`Ok(Err(T))` This is very confusing `Err(T)`
`Err(Ok(T))` This could be very confusing as well
`Err(Err(T))` should be `Err(T)`

Reply:
I've unwrapped the outer `Result` with `JoinError`, which shouldn't happen normally. If it happens, then the main program should crash